File: java/Arrays & Maths/Majority Element.java
public class Solution {
	public int majorityElement(final List<Integer> A) {
	    if (A == null)
	        return -1;
	    int maj = A.get(0);
	    int count = 1;
	    int n = A.size();
	    for (int i = 1; i < n; i++) {
	        if (A.get(i) == maj) {
	            count++;
	        } else if (count == 1) {
	            maj = A.get(i);
	        } else {
	            count--;
	        }
	    }
	    count = 0;
	    for (int i = 0; i < n; i++) {
	        if (A.get(i) == maj)
	            count++;
	    }
	    if (count > n / 2)
	        return maj;
	    return -1;
	}
}

File: python/Arrays & Maths/Majority Element.py
class Solution:
    # @param A : tuple of integers
    # @return an integer
    def majorityElement(self, A):
        majority_idx = 0
        count = 1
        n = len(A)
        for i in range(1,n):
            if A[majority_idx] == A[i]:
                count += 1
            else:
                count -= 1
            if count == 0:
                majority_idx = i
                count = 1
        ret = A[majority_idx]
        count = 0
        for a in A:
            if a == ret:
                count += 1
        if count >= n/2:
            return ret
        return -1